src/main/java/de/tum/in/www1/artemis/service/TextBlockService.java

```diff
@@ -99,4 +103,30 @@ public void deleteForSubmission(TextSubmission textSubmission) {
         textBlockRepository.deleteAllBySubmission_Id(textSubmission.getId());
     }
 
+    /**
+     * Sets number of potential automatic Feedback's for each block belonging to the `Result`'s submission.
+     * This number determines how many other submissions would be affected if the user were to submit a certain block feedback.
+     * For each TextBlock of the submission, this method finds how many other TextBlocks exist in the same cluster.
+     * This number is represented with the `numberOfAffectedSubmissions` field which is set here for each
+     * TextBlock of this submission
+     *
+     * @param result Result for the Submission acting as a reference for the text submission to be searched.
+     */
+    public void setNumberOfAffectedSubmissionsPerBlock(@NotNull Result result) {
+        final TextSubmission textSubmission = (TextSubmission) result.getSubmission();
+        final var blocks = textBlockRepository.findAllWithEagerClusterBySubmissionId(textSubmission.getId());
+        textSubmission.setBlocks(blocks);
+
+        // iterate over blocks of the referenced submission
+        blocks.forEach(block -> {
+            final TextCluster cluster = block.getCluster();
+            final String blockID = block.getId();
+            // if TextBlock is part of a cluster, we find how many other submissions of that cluster it will affect
+            if (cluster != null) {
+                final int numberOfAffectedSubmissions = textBlockRepository.getNumberOfOtherBlocksInCluster(blockID);
```

Comment:
Please fetch this in one query from the database.
If one submission has 50 blocks, this will execute 50 queries.

Here is the SQL query I am thinking of:
```sql
select tb.id, count(distinct tball.id)
from submission s
left join text_block tb on s.id = tb.submission_id
left join text_cluster tc on tb.cluster_id = tc.id
left join text_block tball on tc.id = tball.cluster_id and tball.id != tb.id
where s.id = 2444324
group by tb.id;
```

Reply:
I incorporated the change in the new commit  (https://github.com/ls1intum/Artemis/pull/3456/commits/344a74855fd749be7059702459a047d1f5785fc3) 👍